tools/integration_tests/util/creds_tests/creds.go

```diff
@@ -21,50 +21,75 @@ import (
 	"log"
 	"os"
 	"path"
+	"slices"
+	"strings"
 	"testing"
+	"time"
 
 	"cloud.google.com/go/compute/metadata"
 	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
 	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
 )
 
 const NameOfServiceAccount = "creds-test-gcsfuse"
+const CredentialsBucketName = "gcsfuse-integration-tests-data"
 
-func setPermission(permission string, serviceAccount string) {
-	// Provide permission to the bucket.
-	setup.RunScriptForTestData("../util/creds_tests/testdata/provide_permission.sh", setup.TestBucket(), serviceAccount, permission)
-}
+var WhitelistedGcpProjects = []string{"gcs-fuse-test", "gcs-fuse-test-ml"}
 
 func RunTestsForKeyFileAndGoogleApplicationCredentialsEnvVarSet(testFlagSet [][]string, permission string, m *testing.M) (successCode int) {
+	log.Println("Running credentials tests...")
+
 	// Fetching project-id to get service account id.
 	id, err := metadata.ProjectID()
 	if err != nil {
 		log.Printf("Error in fetching project id: %v", err)
+		return 1
+	}
+	// return if active GCP project is not in whitelisted gcp projects
+	if !slices.Contains(WhitelistedGcpProjects, id) {
+		log.Printf("The active GCP project is not one of: %s. So the credentials test will not run.", strings.Join(WhitelistedGcpProjects, ", "))
+		return 0
 	}
 
 	// Service account id format is name@project-id.iam.gserviceaccount.com
 	serviceAccount := NameOfServiceAccount + "@" + id + ".iam.gserviceaccount.com"
 
-	// Create service account
-	setup.RunScriptForTestData("../util/creds_tests/testdata/create_service_account.sh", NameOfServiceAccount)
+	localKeyFilePath := path.Join(os.Getenv("HOME"), "creds.json")
+	gcsKeyFilePath := path.Join(CredentialsBucketName, id+".json")
 
-	key_file_path := path.Join(os.Getenv("HOME"), "creds.json")
-
-	// Create credentials
-	setup.RunScriptForTestData("../util/creds_tests/testdata/create_key_file.sh", key_file_path, serviceAccount)
+	// Download credentials
+	err = operations.DownloadGcsObject(gcsKeyFilePath, localKeyFilePath)
+	if err != nil {
+		log.Printf("Error while downloading key file %v", err)
```

Comment:
Should't it be log.Fatalf as if key will not download it will exit from that stat itself with exit code 1.

Reply:
As discussed, we are returning successCode=1, so the tests will fail.